File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/punishment/PunishmentDTO.java
package es.jocasolo.competitiveeventsapi.dto.punishment;

import java.io.Serializable;

import es.jocasolo.competitiveeventsapi.dto.DTO;
import es.jocasolo.competitiveeventsapi.dto.image.ImageDTO;
import es.jocasolo.competitiveeventsapi.dto.user.UserLiteDTO;
import es.jocasolo.competitiveeventsapi.enums.score.ScoreSortType;
import lombok.Getter;
import lombok.Setter;

public class PunishmentDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	@Getter
	@Setter
	private Integer id;

	@Getter
	@Setter
	private String title;

	@Getter
	@Setter
	private String description;

	@Getter
	@Setter
	private UserLiteDTO looser;

	@Getter
	@Setter
	private ScoreSortType sortScore;

	@Getter
	@Setter
	private Integer requiredPosition;

	@Getter
	@Setter
	private ImageDTO image;

	@Override
	public String toString() {
		return String.format("RewardDTO [id=%s, title=%s]", id, title);
	}

}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/punishment/PunishmentDTO.kt
package es.jocasolo.competitiveeventsapp.dto.punishment

import com.google.gson.annotations.SerializedName
import es.jocasolo.competitiveeventsapp.dto.user.ImageDTO
import es.jocasolo.competitiveeventsapp.dto.user.UserDTO
import es.jocasolo.competitiveeventsapp.enums.score.ScoreSortType

class PunishmentDTO(
        @SerializedName("id") var id: Int,
        @SerializedName("title") var title: String,
        @SerializedName("description") var description: String?,
        @SerializedName("looser") var looser: UserDTO?,
        @SerializedName("sortScore") var sortScore: ScoreSortType?,
        @SerializedName("requiredPosition") var requiredPosition: Int?,
        @SerializedName("image") var image: ImageDTO?
) {
}